### DISCUSSION TITLE
Runout sensor

### DISCUSSION THREAD

**Discussion by Omar5x** [fra589/grbl-Mega-5X#325] [Q&A]
Is there any possible way to add a runout sensor to mega5x, I'm building a 4 axis tufting machine and i want when the there is no wool thread left the machine will pause.
The thread will move under a mechanical end stop like the one used for 3d printers,  so when there no wool thread left the end-stop will close and the machine will pause is it possible and if yes how can i do it.
Thanks

**Comment by RaphaelDives** [comment]
Hey @Omar5x 
i'm not shure if there is something like that in grbl-mega (since it's not meant for 3d printing), but there is some stuff you can "missuse" for that cases... I would think of the feed hold button for example:
see cpu_map.h line 245  
#define CONTROL_FEED_HOLD_BIT     2  // Pin A10 - RAMPS Aux 2 Port

I'm not shure if there is some option for this mechanism to work (at least I don't think so...).

Cheers 
Raphael

**Reply by Omar5x** [reply]
Hi @RaphaelDives 
Thanks for the fast reply ☺️, so as i understand I'll need to connect an end stop switch to the pin A10 in Aux 2 port and to ground with no modifications to the code, so when the switch is triggered the machine will hold until i start the cycle again,did i understand that correctly ?
Thanks 🙏

**Comment by Omar5x** [comment]
Hi @RaphaelDives
Thanks for the fast reply ☺️, so as i understand I'll need to connect an end stop switch to the pin A10 in Aux 2 port and to ground with no modifications to the code, so when the switch is triggered the machine will hold until i start the cycle again,did i understand that correctly ?
Thanks 🙏

**Comment by RaphaelDives** [comment]
config.h, line 378 states: "Default is normaly open switch between pin to GND," 
So, I would bet that is one way to go.

Since for my understanding this switch is "always active", I would take special care for not being triggered accidently.
(BUT I never tried it myself (yet). The Control Panel for my MPCNC is still on my todo list... :grin:)

**Reply by Omar5x** [reply]
Thank you, I'll try it and hope it works ❤️☺️
